### DISCUSSION TITLE
Case Sensitivity in deserialized dictionaries

### DISCUSSION THREAD

**Discussion by jmoleary** [aaubry/YamlDotNet#881] [Q&A] [answered]
I am replacing some very YAML code with YamlDotNet.  The old code manually reads Yaml into a `Dictionary<string, string>` and parses the values (ugh).   The dictionaries use `StringComparer.OrdinalIgnoreCase` for the keys comparer and the values are raw yaml.  
 
 My first step  using YamlDotNet shows that opening up a file like this yields   `Dictionary<object, object>`  It would be much easier for me to plug this into the rest of my codebase if I could make it instead return `Dictionary<string, object>`, for the top level result and for child dictionaries.     Making it also use `StringComparer.OrdinalIgnoreCase` would be an added bonus, if possible.
 
 Is there a way to achieve this with some sort of configuration option of the Deserializer? 
 
 Here is how I load it now.
 
```
var builder = new DeserializerBuilder().IgnoreUnmatchedProperties();
var yaml = builder.Build();
using var inStream = File.OpenText(YamlFilePath);
if (yaml.Deserialize(inStream) is not Dictionary<object, object> map)
    throw new InvalidOperationException("scan does not have a recognizable map");
```

I see options like  `DeserializerBuilder.WithNamingConvention` and "custom type discriminators" but I don't see how I can use them to achieve this.

**Comment by EdwardCooke** [comment, marked as answer]
The reason it’s an <object, object> is because the yaml spec does not require the keys to be a string. There’s no option to force that to be a string in the library because of that.

**Comment by jmoleary** [comment]
I should add that my eventual solution to this problem was to create a class that implements `IObjectFactory `and specify it in the `DeserializerBuilder.WithObjectFactory` function (during the creation of my Deserializer) to plug it in.  Whenever my ObjectFactory is asked for `Dictionary<object, object>` I instead  make it instead return `Dictionary<string, object>` which gets the property case-insensitive comparer in its constructor.
   Not sure if that is a good solution, it feels like a hack, but it works.
